exp/addons/controllers/clusterresourceset_helpers.go

```diff
@@ -82,31 +82,64 @@ func apply(ctx context.Context, c client.Client, data []byte) error {
 	errList := []error{}
 	sortedObjs := utilresource.SortForCreate(objs)
 	for i := range sortedObjs {
-		if err := applyUnstructured(ctx, c, &objs[i]); err != nil {
+		if err := applyUnstructured(ctx, s, c, &objs[i]); err != nil {
 			errList = append(errList, err)
 		}
 	}
 	return kerrors.NewAggregate(errList)
 }
 
-func applyUnstructured(ctx context.Context, c client.Client, obj *unstructured.Unstructured) error {
+func applyUnstructured(ctx context.Context, s addonsv1.ClusterResourceSetStrategy, c client.Client, obj *unstructured.Unstructured) error {
```

Comment:
nit: I also think its usually like this, but please check other functions here.

```suggestion
func applyUnstructured(ctx context.Context, c client.Client, strategy addonsv1.ClusterResourceSetStrategy, obj *unstructured.Unstructured) error {
```